pyocd/board/board_ids.py

```diff
@@ -61,6 +61,7 @@ def __init__(self, name, target, binary):
     "0324": BoardInfo(  "USB-KW40Z",            "kw40z4",           "l1_kl25z.bin"          ),
     "0400": BoardInfo(  "MAXWSNENV",            "max32600",         "l1_maxwsnenv.bin",     ),
     "0405": BoardInfo(  "MAX32600MBED",         "max32600",         "l1_max32600mbed.bin",  ),
+    "0415": BoardInfo(  "MAX32625MBED",         "max32625",         "max32625mbed.bin",  ),
```

Comment:
Could you add the MAX32625PICO board with ID `0409`? This was requested in #341 some time ago. Even if you don't have a test binary it's probably good to add as long as it's compatible with this PR.

Reply:
The MAX32625PICO is based on MAX32625 HIC, whereas the MAX32625MBED is based on MAX32620 HIC.
MAX32625 HIC is currently not supported in DAPLink, work is in progress though.
I'm not sure how adding it here would help, other than silencing the warning seen in #341.

Reply:
This is for the MAX32625 target, the HIC shouldn't matter. It would enable the MAX32625PICO to automatically select the right target.

Reply:
Cool, adding it soon.

Reply:
I added the support with test binary in my local repo, only the ConnectTest fails on the MAX32625PICO.
Would you suggest adding it now or later?

Reply:
Is the PICO board ID 0416 or 0409? #341 lists it as 0409—did the user report the wrong board, or did the ID change?

Reply:
MAX32625PICO board ID is 0416. It can be used as a target board or a programmer for other boards. A brief description is given [here](https://github.com/mbedmicro/pyOCD/pull/524#discussion_r253230240).
I believe the user in #341 is using MAX32625PICO as a DAPLink adapter to connect to MAX32620FTHR (board ID 0409).

Reply:
Thanks for the explanation!